src/ModuleLocatorInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Dhii\Modular;
+
+/**
+ * Something that can locate and retrieve module configuration.
+ *
+ * Unlike factories and loaders, locators are expected to be configured and consist of all the required data prior to
+ * locating modules. This makes locators more single-purpose specific and disposable than the aforementioned entities.
+ *
+ * @since [*next-version*]
+ */
+interface ModuleLocatorInterface
+{
+    /**
+     * Locates and retrieves module configuration.
+     *
+     * @since [*next-version*]
+     *
+     * @return array The located module configuration as an associative array of module IDs as array keys.
+     *               Array values represent the found module configuration which may be arrays themselves.
+     */
+    public function locate();
```

Comment:
I was thinking no parameters and the instance is assumed to hold configuration. My reasoning was that locators will most likely be specific to a certain "location" (or other determining factors), unlike factories and loaders. I documented this in the interface doc, if you're interesting in reading that.

Also its as you said. It makes it overload-able in descendants. I think this is sufficient, especially since all we can really do at this level is add `$args = array()` as a parameter, which doesn't really signify any particular location strategies.